src/main/java/gregtech/api/recipes/ModHandler.java

```diff
@@ -337,6 +345,83 @@ public static Object finalizeIngredient(Object ingredient) {
         return ingredient;
     }
 
+    public static ItemMaterialInfo getRecyclingIngredients(Object... recipe) {
+        Map<Character, Integer> inputCountMap = new HashMap<>();
+        Map<Material, Long> materialStacksExploded = new HashMap<>();
+
+        int itr = 0;
+        while (recipe[itr] instanceof String) {
+            String s = (String) recipe[itr];
+            for (char c : s.toCharArray()) {
+                if (Character.isLowerCase(c)) continue; // skip tools
+                int count = inputCountMap.getOrDefault(c, 0);
+                inputCountMap.put(c, count + 1);
+            }
+            itr++;
+        }
+
+        char lastChar = ' ';
+        for (int i = itr; i < recipe.length; i++) {
+            Object ingredient = recipe[i];
+
+            // Track the current working ingredient symbol
+            if (ingredient instanceof Character) {
+                lastChar = (char) ingredient;
+                continue;
```

Comment:
Why does this continue after finding the current working symbol? Shouldn't it allow following down the loop to get to the other parts instead of `continue`?

Reply:
It's just how it ended up laying out, alternatively I could iterate over `i += 2` in the loop, but either work fine